watchmal/engine/engine_classifier.py

```diff
@@ -525,7 +525,7 @@ def restore_state_from_file(self, weight_file):
             self.model_accs.load_state_dict(checkpoint['state_dict'])
             
             # if optim is provided, load the state of the optim
-            if hasattr(self, 'optimizer'):
+            if getattr(self, 'optimizer') is not None:
```

Comment:
I think this can just be
`if self.optimizer is not None:`
which reads a bit nicer. Could you check?

Reply:
Maybe my version would fail if there is no `optimizer` attribute, but then the original code would have worked.